src/libcore/slice.rs

```diff
@@ -684,13 +666,11 @@ macro_rules! iterator {
             fn next(&mut self) -> Option<$elem> {
                 // could be implemented with slices, but this avoids bounds checks
                 unsafe {
-                    ::intrinsics::assume(!self.ptr.is_null());
-                    ::intrinsics::assume(!self.end.is_null());
```

Comment:
I'm of the impression that such an object size and placement would be illegal/impossible. If it isn't, we should make sure it is in safe rustc. We can't safely index/address all bytes of that object otherwise.

GEP is our `.offset()`.

> If the GEP has the inbounds keyword, the result value is undefined (a “trap value”) if the GEP overflows (i.e. wraps around the end of the address space).

http://llvm.org/docs/GetElementPtr.html#what-happens-if-a-gep-computation-overflows